src/main/java/net/neoforged/neoforge/common/extensions/IItemExtension.java

```diff
@@ -591,15 +592,16 @@ default boolean canDisableShield(ItemStack stack, ItemStack shield, LivingEntity
     }
 
     /**
-     * @return the fuel burn time for this itemStack in a furnace. Return 0 to make
+     * @return the fuel burn time for this item stack in a furnace. Return 0 to make
      *         it not act as a fuel. Return -1 to let the default vanilla logic
      *         decide.
      * @apiNote This method takes precedence over the {@link net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps#FURNACE_FUELS data map}.
-     *          However, you should use the datamap unless necessary (i.e. NBT-based burn times) so that users can configure burn times.
+     *          However, you should use the data map unless necessary (i.e. NBT-based burn times) so that users can configure burn times.
      */
     @ApiStatus.OverrideOnly
     default int getBurnTime(ItemStack itemStack, @Nullable RecipeType<?> recipeType) {
-        return -1;
+        var fuel = self().builtInRegistryHolder().getData(NeoForgeDataMaps.FURNACE_FUELS);
```

Comment:
Don't use `var` here, as the type of `fuel` is not explicitly clear without an IDE present.

Reply:
fixed